### ISSUE TITLE
Investigate GPU-side computation of time steps.

### ISSUE THREAD

**Issue by thorstenhater** [arbor-sim/arbor#1371]
This came up in a discussion over #1357, where `vec_t_to` was remove since redundant.
Historically the per-cv-timestep was computed on the fly instead of precalculated into a
vector. @halfflat claimed that had been faster. Also, it had been claimed that not having 
per-cv-dt could be even faster (on the GPU only) in different contexts.

We should investigate if one of the following alternatives is clearly superior in terms of performance
- [ ] Current model: Pre-compute `dt_intdom` and `dt_cv`
- [ ] Compute `dt` on the fly as needed
- [x] Just use a fixed `dt`: Out of scope, has other implications.

The experiment should use simple mechanisms and a complicated cell to emphasize the
impact of the matrix solver.

It would be nice to have this before finalising the ABI #1335.

**Comment by halfflat** [comment]
The fixed vs flexible dt is a related but different investigation: we have flexible dt steps so that we can accommodate exact spike delivery times, and provides a structure on top of which we might implement variable time stepping. The simulation justification for exact spike time delivery is anecdotal: the anecdote is that with binned spikes, some networks can show spurious correlations in spiking activity. There is definitely a cost, in terms of mopping up remaining time steps at the end of an epoch, but this cost has not been well quantified. I think this should be a separate issue.

The API could, in principle, support both mechanism-side calculation of dt and library-side: if dt pointer is not null use it, otherwise use time_to and time. This could allow the investigation to proceed in parallel with the ABI work.

**Comment by thorstenhater** [comment]
Option 2 is also currently not viable, as the pre-computed is needed to expose `time` to mechanisms. If that is no longer needed, we should revisit this maybe.
See #1252
